src/ng-sentry.common.ts

```diff
@@ -0,0 +1,158 @@
+import { getVersionName, getVersionCode } from 'nativescript-appversion';
+import * as platform from 'tns-core-modules/platform';
+import * as http from 'tns-core-modules/http';
+import { setString, getString, remove } from 'tns-core-modules/application-settings';
+import * as moment from 'moment/moment';
+
+export interface KeyValue<T> { [key: string]: T; }
+
+export class Common {
+    private static instance: Common = new Common();
+
+    private maxAmountOfBreadcrumbs: number = 50;
+    private endPoint: string;
+    private readonly storageKey: string = 'sentryCrashesQueue';
+
+    private breadcrumbs: Array<any> = [];
+    private versionName: string = '';
+    private versionCode: string = '';
+    private environment: string = 'DEVELOPMENT';
+
+    constructor() {
+        if (Common.getInstance()) {
+            throw new Error("Error: Instance failed: Use LiveEngage.getInstance() instead of new.");
+        }
+
+        Common.instance = this;
+
+        // Fix for Android timing issue
+        setTimeout(() => {
+            this.setVersionName();
+            this.setVersionCode();
+        });
+    }
+
+    static getInstance() {
+        return Common.instance;
+    }
+
+    public setCredentials(sentryId: string, sentryKey: string, environment?: string) {
+        this.endPoint = 'https://sentry.io/api/' + sentryId + '/store/?sentry_version=7&sentry_client=raven-js%2F3.20.1&sentry_key=' + sentryKey;
+        if (environment) {
+            this.environment = environment;
+        }
+    }
+
+    public setMaxAmountOfBreadcrumbs(amount: number) {
+        this.maxAmountOfBreadcrumbs = amount;
+    }
+
+    public saveBreadcrumb(message: string, category: string, properties?: KeyValue<string>): void {
+        const crumb = {
+            timestamp: moment(),
+            message,
+            category,
+            data: properties
+        };
+        this.breadcrumbs.push(crumb);
+        if (this.breadcrumbs.length > this.maxAmountOfBreadcrumbs) {
+            this.breadcrumbs.shift();
+        }
+    }
+
+    public saveCrash(errorMessage: string, errorDetails: string): void {
+        this.saveBreadcrumb('' + errorDetails, 'crash start');
+
+        const crashData = {
+            project: '253147',
+            logger: 'nativescript',
+            platform: 'javascript',
+            message: '' + errorMessage,
+            level: 'error',
+            tags: {
+                device_type: platform.device.deviceType,
+                device_lang: platform.device.language,
+                app_version: this.versionCode,
+                trace_category: 'Error'
+
+            },
+            breadcrumbs: {
+                values: this.breadcrumbs
+            },
+            environment: this.environment,
+            release: this.versionName,
+            contexts: {
+                device: {
+                    family: platform.device.manufacturer,
```

Comment:
Is platform.device always available, if not this way the plugin itself will crash?

Reply:
According to the documentation (https://docs.nativescript.org/cookbook/platform) the properties can be accessed without a check if they're available.